apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,745 @@
+    * @param _representative Address of the representative to be allowed on behalf of the sender. Use the zero address for none.
+    */
+    function setRepresentative(address _representative) external isInitialized {
+        representatives[msg.sender] = _representative;
+        emit ChangeRepresentative(msg.sender, _representative);
+    }
+
+    /**
+    * @notice Create a new vote about "`_metadata`"
+    * @param _executionScript EVM script to be executed on approval
+    * @param _metadata Vote metadata
+    * @param _castVote Whether to also cast newly created vote
+    * @return voteId id for newly created vote
+    */
+    function newVote(bytes _executionScript, string _metadata, bool _castVote)
+        external
+        authP(CREATE_VOTES_ROLE, arr(msg.sender))
+        returns (uint256 voteId)
+    {
+        return _newVote(_executionScript, _metadata, _castVote);
+    }
+
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    * @param _supports Whether voter supports the vote
+    */
+    function vote(uint256 _voteId, bool _supports) external voteExists(_voteId) {
+        require(_canVote(votes[_voteId], msg.sender), ERROR_CAN_NOT_VOTE);
+        _vote(_voteId, _supports, msg.sender);
+    }
+
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of a set of voters
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voters Addresses of the voters voting on behalf of
+    * @param _voteId Id for vote
+    * @param _supports Whether the representative supports the vote
+    */
+    function voteOnBehalfOf(address[] _voters, uint256 _voteId, bool _supports) external voteExists(_voteId) {
+        require(_voters.length <= MAX_VOTES_DELEGATION_SET_LENGTH, ERROR_DELEGATES_EXCEEDS_MAX_LEN);
+        _voteOnBehalfOf(_voters, _voteId, _supports);
+    }
+
+    /**
+    * @notice Execute vote #`_voteId`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    */
+    function executeVote(uint256 _voteId) external voteExists(_voteId) {
+        _executeVote(_voteId);
+    }
+
+    // Forwarding fns
+
+    /**
+    * @notice Creates a vote to execute the desired action, and casts a support vote if possible
+    * @dev IForwarder interface conformance
+    *      Disputable apps require to be the initial step in the forwarding chain
+    * @param _evmScript Start vote with script
+    */
+    function forward(bytes _evmScript) public {
```

Comment:
I assume we'll be updating this in the future with the bytes?

Reply:
Yes, when the new forwarding interface comes :)